test/unit/image/p5.Image.js

```diff
@@ -49,4 +49,70 @@ suite('p5.Image', function() {
       assert.strictEqual(img.height, 30);
     });
   });
+
+  suite('p5.Image.prototype.mask', function() {
+    test('it should mask the image', function() {
+      let img = myp5.createImage(10, 10);
+      img.loadPixels();
+      for (let i = 0; i < img.height; i++) {
+        for (let j = 0; j < img.width; j++) {
+          let alpha = i < 5 ? 255 : 0;
+          img.set(i, j, myp5.color(0, 0, 0, alpha));
+        }
+      }
+      img.updatePixels();
+
+      let mask = myp5.createImage(10, 10);
+      mask.loadPixels();
+      for (let i = 0; i < mask.width; i++) {
+        for (let j = 0; j < mask.height; j++) {
+          let alpha = j < 5 ? 255 : 0;
+          mask.set(i, j, myp5.color(0, 0, 0, alpha));
+        }
+      }
+      mask.updatePixels();
+
+      img.mask(mask);
+      img.loadPixels();
+      for (let i = 0; i < img.width; i++) {
+        for (let j = 0; j < img.height; j++) {
+          let alpha = i < 5 && j < 5 ? 255 : 0;
+          assert.strictEqual(img.get(i, j)[3], alpha);
+        }
+      }
+    });
+
+    test('it should mask the animated gif image', function() {
+      const imagePath = 'unit/assets/nyan_cat.gif';
+      return new Promise(function(resolve, reject) {
+        myp5.loadImage(imagePath, resolve, reject);
+      }).then(function(img) {
+        let mask = myp5.createImage(img.width, img.height);
+        mask.loadPixels();
+        for (let i = 0; i < mask.width; i++) {
+          for (let j = 0; j < mask.height; j++) {
+            const alpha = j < img.height < 2 ? 255 : 0;
+            mask.set(i, j, myp5.color(0, 0, 0, alpha));
+          }
+        }
+        mask.updatePixels();
+
+        img.mask(mask);
```

Comment:
It would be great to check the current pixels here too, just like in the static image test above

Reply:
Added a check for the current pixels.